data_steward/constants/validation/participants/validate.py

```diff
@@ -229,62 +107,145 @@
                 ELSE '{{missing_ehr}}'
             END AS name
         FROM normalized_rdr_name, normalized_ehr_name 
-
     ));
 """)
 
 CREATE_STREET_COMPARISON_FUNCTION = JINJA_ENV.from_string("""
 CREATE FUNCTION IF NOT EXISTS
-  `{{project_id}}.{{drc_dataset_id}}.CompareStreetAddress`(rdr_street string, ehr_street string)
-  RETURNS string AS ((
-    {{street_with_clause}}
+  `{{project_id}}.{{drc_dataset_id}}.CompareStreet`(rdr_street string, ehr_street string)
+  RETURNS string 
+  AS ((
+    WITH 
+      normalized_rdr AS ({{rdr_street}}),
+      normalized_ehr AS ({{ehr_street}})
     SELECT
       CASE
-        WHEN normalized_rdr_street.rdr_street = normalized_ehr_street.ehr_street THEN '{{match}}'
-        WHEN normalized_rdr_street.rdr_street IS NOT NULL AND normalized_ehr_street.ehr_street IS NOT NULL THEN '{{no_match}}'
-        WHEN normalized_rdr_street.rdr_street IS NULL THEN '{{missing_rdr}}'
-        ELSE '{{missing_ehr}}'
+        WHEN rdr_street IS NULL THEN '{{missing_rdr}}'
+        WHEN ehr_street IS NULL THEN '{{missing_ehr}}'
+        WHEN normalized_rdr.street = normalized_ehr.street THEN '{{match}}'
+        ELSE '{{no_match}}'
       END AS street
-    FROM normalized_rdr_street, normalized_ehr_street));
+    FROM normalized_rdr, normalized_ehr
+  ));
 """)
 
 CREATE_CITY_COMPARISON_FUNCTION = JINJA_ENV.from_string("""
 CREATE FUNCTION IF NOT EXISTS
   `{{project_id}}.{{drc_dataset_id}}.CompareCity`(rdr_city string, ehr_city string)
   RETURNS string AS ((
-    {{city_with_clause}}
+    WITH 
+      normalized_rdr AS ({{rdr_city}}),
+      normalized_ehr AS ({{ehr_city}})    
     SELECT
       CASE
-        WHEN normalized_rdr_city.rdr_city = normalized_ehr_city.ehr_city THEN '{{match}}'
-        WHEN normalized_rdr_city.rdr_city IS NOT NULL AND normalized_ehr_city.ehr_city IS NOT NULL THEN '{{no_match}}'
-        WHEN normalized_rdr_city.rdr_city IS NULL THEN '{{missing_rdr}}'
-        ELSE '{{missing_ehr}}'
+        WHEN rdr_city IS NULL THEN '{{missing_rdr}}'
+        WHEN ehr_city IS NULL THEN '{{missing_ehr}}'
+        WHEN normalized_rdr.city = normalized_ehr.city THEN '{{match}}'
+        ELSE '{{no_match}}'
       END AS city
-    FROM normalized_rdr_city, normalized_ehr_city));
+    FROM normalized_rdr, normalized_ehr));
+""")
+
+NORMALIZED_STREET = JINJA_ENV.from_string("""
+    WITH address_abbreviations AS (
+        SELECT *
+        FROM UNNEST(ARRAY<STRUCT<abbreviation STRING, expansion STRING>>[
+            {{abbreviation_tuples}}
+        ])
+    ),
+    remove_commas_and_periods AS (
+        SELECT 'dummy_id' as id, REGEXP_REPLACE(coalesce({{street}}, ''), '[,.]', '') as address
```

Comment:
I'd suggest using something like row_number over() as the dummy_id.

Reply:
I think it is better to keep it this way, because `CompareStreet` (the function that uses this clause) reads only one row at a time, not the whole table. So, `row_number` makes it look more complex but does not improve the logic much.

Reply:
I see, I thought this would be necessary to distinguish different addresses but it looks like this is only processing one address at a time.